frontend/src/views/components/Legend/LegendTimeline/LegendTimeline.tsx

```diff
@@ -0,0 +1,29 @@
+import { connect } from 'react-redux';
+import { setDate } from 'state/modules/layers';
+import LegendTimelineComponent from './LegendTimeline.component';
+import { timeFormat, timeParse } from 'd3-time-format';
+import type { Timeline } from 'types/layer';
+
+const mapDispatchToProps = {
+  setDate,
+};
+
+interface LegendTimelineContainerProps {
+  setDate: (layerId: string, date: Date) => void;
+  layerId: string;
+  layerName: string;
+  selectedDate: string;
+  timeline: Timeline;
+}
+
+const LegendTimelineContainer = (props: LegendTimelineContainerProps) => {
+  const { setDate: setDateAction, layerId, selectedDate, timeline } = props;
+  const { format, defaultDate } = timeline;
+  const setDate = (date: Date) => setDateAction(layerId, timeFormat(format)(date));
+  const parseDate = timeParse(format);
+  const selected = selectedDate ? new Date(parseDate(selectedDate)) : defaultDate;
```

Comment:
Here I also think it's incorrect. `timeFormat` and `parseDate` should be used for _displaying_ the date only.